v4/weaviate/graphql/neartextbuilder.go

```diff
@@ -1,11 +1,72 @@
 package graphql
 
 import (
-	"encoding/json"
 	"fmt"
 	"strings"
 )
 
+// fldMover is a type representing field names of a move sub query
+type fldMover string
+
+const (
+	fldMoverConcepts fldMover = "concepts"
+	fldMoverForce    fldMover = "force"
+	fldMoverID       fldMover = "id"
+	fldMoverBeacon   fldMover = "beacon"
+	fldMoverObjects  fldMover = "objects"
+)
+
+// MoveParameters to fine tune Explore queries
+type MoveParameters struct {
+	// Concepts that should be used as base for the movement operation
+	Concepts []string
+	// Force to be applied in the movement operation
+	Force float32
+	// Movers is a list of movers used to adjust the serach direction
+	Movers []MoverObject
+}
+
+func (m *MoveParameters) String() string {
+	concepts := marshalStrings(m.Concepts)
```

Comment:
I would remove the `marshalStrings` method and make the code simpler and better to read by refactoring for example to this version:
```
clause := []string{}
if len(m.Concepts) > 0 {
    concepts := make([]string, len(m.Concepts))
    for i := range m.Concepts {
	concepts[i] = fmt.Sprintf("\"%s\"", m.Concepts[i])
   }
   clause = append(clause, fmt.Sprintf("concepts:[%s]", strings.Join(concepts, ", ")))
}
```

Reply:
I think on the contrary `marshalStrings` makes code simpler and better to read and refactor:

I can think at leat of two reasons why this is the case:
1. `marshalStrings` has the single responsibility of marshalling slices of strings which results in better readability and testability.
2. This function is need in different places which removes redundancy. In contrast the proposed solution would result in redundant and hard to test code.

This function replaces ` concepts, _ := json.Marshal(m.Concepts)` which as @parkerduckworth has noticed ignore the error.